### ISSUE TITLE
Make PN note accessible on all cache details tabs

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#12729]
> Would it help if the personal notes would be available on the variable tab too? So you can use it for copy/pasting?

Yes sounds like a good idea. 

BTW, I just noticed  that we already have a "Personal note" shortcut in the actionbar overflow menu. Instead of blocking the whole screen with a dialog, we could use a so called "Standart bottom sheet". 

Material.io defines 3 different bottom sheet types:
> -    Standard bottom sheets display content that complements the screen’s primary content. They remain visible while users interact with the primary content.
> -    Modal bottom sheets are an alternative to inline menus or simple dialogs on mobile and provide room for additional items, longer descriptions, and iconography. They must be dismissed in order to interact with the underlying content.
> -    Expanding bottom sheets provide a small, collapsed surface that can be expanded by the user to access a key feature or task. They offer the persistent access of a standard sheet with the space and focus of a modal sheet.
> 
>

_Originally posted by @fm-sys in https://github.com/cgeo/cgeo/issues/12718#issuecomment-1037911071_

**Comment by ztNFny** [comment]
related to https://github.com/cgeo/cgeo/issues/8928

**Comment by fm-sys** [comment]
> related to #8928

Not really the same, the other one is about the map cache popup...

**Comment by fm-sys** [comment]
We cannot use a dialog, if we want to allow users to interact with the bottom sheet and the other content at the same time. A dialog gets closed when clicking outside.

**Comment by eddiemuc** [comment]
I thought about it, and maybe this would not solve the original request (which was having some sort of a "copy&paste zone" in variable tab). Reason is that the current workflow of PN requires every edit of it to be eplicitely confirmed (so certain routines can run in background e.g. extraction of waypoins). This would make it hard/impossible to place PN in an edit field which is editable "on the fly" (without explicit "ok" button to be used when done), because such processing must then happen on every typed char.

**Comment by moving-bits** [comment]
> > Would it help if the personal notes would be available on the variable tab too? So you can use it for copy/pasting?

Actually, personal notes are already available on every cache details tab - just use the three dot menu => "personal note". A full screen dialog with the personal note opens, so you can copy/paste some text, and if you close the dialog, you will return to the place where you have been before.

**Comment by dcjkfgdjhd** [comment]
I have long thought that the PN should simply have ITS OWN TAB. It is too frequently used to be at the bottom of the listing text, and also I don't really see why it logically would belong there.

**Comment by ztNFny** [comment]
> I have long thought that the PN should simply have ITS OWN TAB. It is too frequently used to be at the bottom of the listing text, and also I don't really see why it logically would belong there.

In terms of reading the PN it makes sense to have it next to the hint as both fields will give you guidance in finding the cache. E.g i always copy the text from checker to note.
That's why I'm the popup if you click the show hint button both hint and note are shown.
Moving it to it's own tab would require more clicks for a very common situation - that both hint and note are checked.

**Comment by dcjkfgdjhd** [comment]
Hmm, not sure whether I agree to the claim that one often needs PN and hint at the same time. 

Usually, one uses the PN to write down variable values that one collects during a multicache, and during that time you normally don't need the hint. The hint usually only become relevant after you have already calculated the final coordinates to find the final cache, and at that time, you don't need the PN any more.

But of course I would not be opposed to also move the hint to the same new tab as the PN. Having the hint accessible without having to scroll to the bottom of the listing certainly would also be an improvement.

**Comment by ztNFny** [comment]
There's more than 1 way of using cgeo. You're talking only about multi whereas my example above was for a mystery.
Even for multis i do it completely different to you, i add the variables as variables and only once the cache is found export to note 

But even for your usecase i'd argue it's useful to have listing, hint and note together: you read the listing to get the steps how to calculate the next stage, use the hint to find it, then use note to document that.

**Comment by dcjkfgdjhd** [comment]
> you read the listing to get the steps how to calculate the next stage, use the hint to find it, then use note to document that.

Yes, and for that it is quite annoying to have to scroll up and down through the listing all the time. 

If, OTOH, the PN would be its own tab, then the scrolling location in the listing would be saved when you swipe left and right between listing view, and PN/Hint tab.

**Comment by ztNFny** [comment]
> If, OTOH, the PN would be its own tab, then the scrolling location in the listing would be saved when you swipe left and right between listing view and PN/Hint tab.

True, it'd save some time in those situations. But it'd cost more time in situations where you just want to read the listing, cache and note.
For my personal usage profile I'd probably loose 10 times as much time as I'd gain.

And it causes further problems:
- Scrolling to logs/images/... tabs requires additional swipes (if note would be next to listing - but if it isn't then even your example will not save any time) and thus lost time
- It changes established workflows, breaking any muscle-memory users have
